src/include/daos/object.h

```diff
@@ -176,13 +176,40 @@ struct daos_obj_layout {
 #define DAOS_TGT_IGNORE		((d_rank_t)-1)
 
 enum daos_tgt_flags {
-	/* When leader forward IO RPC to non-leaders, delay the target until the others replied. */
+	/*
+	 * When leader forward IO RPC to non-leaders, delay the target until the others replied.
+	 *
+	 * For old client (2.0 or older), it will not set such flag when trigger related IO RPC,
+	 * then it may get trouble for conditional check during rebuild. To avoid that, need to
+	 * upgrade DAOS (both client and server) to at least 2.2 or newer. See DAOS-10204.
+	 */
 	DTF_DELAY_FORWARD	= (1 << 0),
-	/* When leader forward IO RPC to non-leaders, reassemble related sub request. */
+	/*
+	 * When leader forward IO RPC to non-leaders, reassemble related sub request.
+	 *
+	 * For old client (2.0 or older), it will not set such flag when trigger related CPD RPC
+	 * (for punch object across multiple RDGs), then it may get trouble during OSA. To avoid
+	 * that, need to upgrade DAOS (both client and server) to at least 2.2 or newer.
```

Comment:
im sorry i don't think i quite understand this. i will leave to @wangdi1  and @johannlombardi to confirm.
are you saying a 2.0 client can cause problems on the server side? what is "trouble" here? if that can cause assertions, segfaults, or server exclusions, then i do not think this will be acceptable.

Reply:
As Di explained in other comment, we do not support OSA until 2.4. So here, this comment is not accurate. I will remove the stale comment when have other chance to refresh the patch.